app/models/concerns/host_searching.rb

```diff
@@ -43,28 +43,26 @@ module HostSearching
       with_policy.or(with_test_results)
     }
 
-    scope :os_major_version, lambda { |version, equal = true|
-      condition = ['system_profile @> ?',
-                   { operating_system: { major: version.to_i } }.to_json]
-      equal && where(*condition) || where.not(*condition)
+    scope :os_major_version, lambda { |versions, equal = true|
+      where(::Host.os_version_query(:major, versions, equal))
     }
 
-    scope :os_minor_version, lambda { |version, equal = true|
-      condition = ['system_profile @> ?',
-                   { operating_system: { minor: version.to_i } }.to_json]
-      equal && where(*condition) || where.not(*condition)
+    scope :os_minor_version, lambda { |versions, equal = true|
+      where(::Host.os_version_query(:minor, versions, equal))
     }
   end
 
   # class methods for Host searching
   module ClassMethods
     def filter_os_major_version(_filter, operator, value)
-      hosts = ::Host.os_major_version(value, operator == '=')
+      values = value.split(',').map(&:to_i)
```

Comment:
This would likely raise HTTP 500 if user provides invalid input. Does the lib provide some input validation?

Reply:
The parser only lets through strings, any string can be splitted, worst case the result will be an empty string that converted to an integer returns a 0. Which could match 8.0 when looking for a minor one, but I guess that's okay.

Reply:
I had the `to_i` conversion in mind.